forwardcookie.go

```diff
@@ -55,77 +47,51 @@ func New(ctx context.Context, next http.Handler, config *Config, name string) (h
 }
 
 func (e *ForwardCookie) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
-	isFetchReq := true
-	// if the incoming request contains cookies
-	if len(req.Cookies()) > 0 {
-		// make sure that those cookies won't be added two-fold
-		for _, name := range e.cookies {
-			cookie, err := req.Cookie(name)
-			if err != nil {
-				isFetchReq = false
-			}
-			if cookie != nil {
-				isFetchReq = false
-			}
-		}
+	fetchReq, err := http.NewRequest(http.MethodGet, e.addr, nil)
+	if err != nil {
+		log.Printf("%s", err)
+		return
 	}
-	if isFetchReq {
-		fetchReq, err := http.NewRequest(http.MethodGet, e.addr, nil)
-		if err != nil {
-			log.Fatalf(fmt.Sprintf("%s", err))
-			return
-		}
 
-		AddHeaders(fetchReq, req, e)
-		AddParameters(fetchReq, req, e)
+	addCookiesFromRequest(fetchReq, req, e)
+	addHeaders(fetchReq, req, e)
+	addParameters(fetchReq, req, e)
 
-		client := http.Client{}
-		forwardResponse, err := client.Do(fetchReq)
-		if err != nil {
-			log.Fatalf(fmt.Sprintf("%s", err))
-			return
-		}
+	forwardResponse, err := http.DefaultClient.Do(fetchReq)
+	if err != nil {
+		log.Printf("%s", err)
+		return
+	}
```

Comment:
Consider adding retries here with exponential backoff. I think IDM isn't always the most reliable service around and since auth is critical for the functionality of any service it might be worth adding retires here to reduce the risk of running into auth issues if IDM struggles temporarily or something similar.

Reply:
Agree we should look into this before going prod.